registry/remote/example_test.go

```diff
@@ -310,6 +330,35 @@ func ExampleRepository_Fetch_manifestByDigest() {
 	// Example manifest content
 }
 
+func ExampleRepository_Fetch_referenceManifest() {
```

Comment:
In this example, we want to demonstrate how to download the referrer (exampleReferenceManifest) of a given manifest.
The steps would be:
1. Resolve the image manifest by tag
2. Discover the referrers of the resolved manifest by calling the [Referrers()](https://github.com/oras-project/oras-go/blob/main/registry/remote/repository.go#L339-L356) (See [Referrers API](https://github.com/oras-project/artifacts-spec/blob/main/manifest-referrers-api.md) for more details)
3. Fetch the referrer (exampleReferenceManifest)

Reply:
Added such an example.